packages/lesswrong/components/ea-forum/users/EAGApplicationImportForm.tsx

```diff
@@ -0,0 +1,702 @@
+          biography: {
+            markdownValue: bio,
+            ckEditorValue: markdownToHtmlSimple(bio)
+          },
+          howOthersCanHelpMe: {
+            markdownValue: howOthersCanHelpMe,
+            ckEditorValue: markdownToHtmlSimple(howOthersCanHelpMe)
+          },
+          howICanHelpOthers: {
+            markdownValue: howICanHelpOthers,
+            ckEditorValue: markdownToHtmlSimple(howICanHelpOthers)
+          },
+          mapLocation: eagData.Your_nearest_city,
+          linkedinProfileURL: eagData.LinkedIn_URL || eagData.LinkedIn_profile_summary
+        })
+      }
+      setFormLoading(false)
+    }
+    // eslint-disable-next-line no-console
+    fetchData().catch(console.error)
+  }, [])
+  
+  const formFields = [
+    'jobTitle',
+    'organization',
+    'careerStage',
+    'biography',
+    'howOthersCanHelpMe',
+    'howICanHelpOthers',
+    // 'organizerOfGroupIds', // TODO: implement later - for the first release I decided this wasn't worth the effort to include
+    'mapLocation',
+    'linkedinProfileURL',
+  ]
+  // formValues holds the state of the form EXCEPT for CKEditor fields, which have their own state
+  const [formValues, setFormValues] = useState(pick(currentUser, formFields))
+
+  // these are used to access CKEditor fields, to copy over the imported data
+  const biographyRef = useRef<EditorFormComponentRefType>(null)
+  const howOthersCanHelpMeRef = useRef<EditorFormComponentRefType>(null)
+  const howICanHelpOthersRef = useRef<EditorFormComponentRefType>(null)
+  // CKEditor fields use this to insert their data before we submit the form
+  const submitFormCallbacks = useRef<Array<Function>>([])
+  // CKEditor fields use this to clear local storage after successfully submitting the form
+  const successFormCallbacks = useRef<Array<Function>>([])
+  
+  const updateCurrentUser = useUpdateCurrentUser()
+  
+  const importMapLocation = async () => {
+    if (mapsLoaded && importedData?.mapLocation) {
+      // get a list of matching Google locations for the current lat/lng (reverse geocoding)
+      const geocoder = new googleMaps.Geocoder()
+      const geocodingResponse = await geocoder.geocode({
+        address: importedData.mapLocation
+      })
+      const results = geocodingResponse?.results
+      if (results?.length) {
+        return pickBestReverseGeocodingResult(results)
+      }
+    }
+    return null
+  }
+  
+  const importLinkedinProfileURL = () => {
+    // try to get the relevant snippet of the linkedin profile URL
+    const matches = importedData?.linkedinProfileURL?.match(/linkedin\.com\/in\/(\S+)/i)
+    return matches?.length ? matches[1] : ''
+  }
+  
+  const handleCopyAll = async (e) => {
+    e.preventDefault()
+    if (!importedData) return
+
+    // @ts-ignore TODO
+    setFormValues({
+      ...importedData,
+      mapLocation: await importMapLocation(),
+      linkedinProfileURL: importLinkedinProfileURL()
+    })
```

Comment:
There's a type mismatch here, since `importedData` is missing some info for CKEditor fields that `UsersCurrent` has. I'm tempted to just ignore that, since the functionality is still correct, but let me know if you know of a way to properly type this.